From e8cd12c58270f44c7f94b2c3a1299c18acf41ab4 Mon Sep 17 00:00:00 2001
From: "Ai, Ting" <ting.a.ai@intel.com>
Date: Thu, 13 Jan 2022 16:51:29 +0800
Subject: [PATCH] Allow wechat to access 9p share folder

Fix the issue that wechat can not read files in host sharefolder

Tracked-On: OAM-100821
Signed-off-by: Ai, Ting <ting.a.ai@intel.com>
---
 aafd/file.te        | 1 -
 aafd/genfs_contexts | 1 -
 aafd/logwrapper.te  | 2 ++
 3 files changed, 2 insertions(+), 2 deletions(-)
 delete mode 100644 aafd/file.te

diff --git a/aafd/file.te b/aafd/file.te
deleted file mode 100644
index 65ba89f..0000000
--- a/aafd/file.te
+++ /dev/null
@@ -1 +0,0 @@
-type p9fs2, fs_type, mlstrustedobject;
diff --git a/aafd/genfs_contexts b/aafd/genfs_contexts
index 337bf7b..2b00f2e 100644
--- a/aafd/genfs_contexts
+++ b/aafd/genfs_contexts
@@ -1,2 +1 @@
-genfscon 9p / u:object_r:p9fs2:s0
 genfscon sysfs /class/udc u:object_r:sysfs_usb:s0
diff --git a/aafd/logwrapper.te b/aafd/logwrapper.te
index 6d40ff8..53433a0 100644
--- a/aafd/logwrapper.te
+++ b/aafd/logwrapper.te
@@ -25,6 +25,8 @@ allow logwrapper p9fs2:dir r_dir_perms;
 allow platform_app p9fs2:file r_file_perms;
 allow platform_app p9fs2:dir r_dir_perms;
 allow untrusted_app_29 p9fs2:file r_file_perms;
+allow untrusted_app_29 p9fs2:file create_file_perms;
+allow untrusted_app_29 p9fs2:dir rw_dir_perms;
 
 set_prop(logwrapper, vendor_suspend_prop)
 set_prop(logwrapper, vendor_intel_ipaddr_prop)
-- 
2.25.1

